Modules/timemodule.c

```diff
@@ -2122,36 +2162,28 @@ pysleep(_PyTime_t secs)
             Py_BEGIN_ALLOW_THREADS
             Sleep(ul_millis);
             Py_END_ALLOW_THREADS
+            ret = 0;
             break;
         }
 
-        hInterruptEvent = _PyOS_SigintEvent();
-        ResetEvent(hInterruptEvent);
-
         Py_BEGIN_ALLOW_THREADS
-        rc = WaitForSingleObjectEx(hInterruptEvent, ul_millis, FALSE);
+        rc = WaitForSingleObjectEx(hTimer, ul_millis, FALSE);
```

Comment:
That's wrong. You should wait for hInterruptEvent. Otherwise, you will not get notified when Python is interrupted by a signal (CTRL+C). You should use WaitForMultipleObjectsEx() for example.

Reply:
WaitForMultipleObjectsEx() was added